core/domain/stats_jobs_one_off.py

```diff
@@ -227,6 +230,20 @@ def reduce(exp_id, stringified_values):
             elif value['event_type'] == feconf.EVENT_TYPE_STATE_HIT:
                 state_name = value['state_name']
                 session_id = value['session_id']
+                event_ts = value['created_on']
+
+                # Some state hit events have version as None. In these cases,
+                # we identify the version by comparing the event timestamp with
+                # the snapshot timestamps.
+                if version is None:
+                    # If the event timestamp is greater than the max snapshot
+                    # timetamp, then it is the latest version.
+                    if event_ts > max(snapshot_timestamps):
+                        version = len(snapshot_timestamps)
+                    else:
+                        for index, timestamp in enumerate(snapshot_timestamps):
+                            if event_ts < timestamp:
+                                version = index
 
```

Comment:
I think you should do this a bit more defensively. Once you've determined the version using this approach, check to see whether the state_name for the event is in the correct exploration object. If it is, great. If not, ignore the event, but log lots of stuff (exp id, state name, state hit event id, etc.) so we can debug.

Reply:
Done.